### PR TITLE
Cargo.lock: Run cargo update

### PR DESCRIPTION
**Pull request by mxinden**
I suggest merging this large dependency update into Substrate `master`, like we did for Polkadot `master` with https://github.com/paritytech/polkadot/pull/1946, before enabling Dependabot as proposed via https://github.com/paritytech/substrate/pull/7509.

What do you think? Objections?

### CODE REVIEW

**Comment by ordian** [comment]
There are a few dependencies that can be upgraded in `Cargo.toml` as well (potentially breaking changes):
<details>
<summary> List </summary>

hyper
nix
platforms
parking_lot
bytes
impl-trait-for-tuples
rpassword
tokio
wasmi
parity-wasm
pin-project
directories
linregress
paste
evm
sha3
num-bigint
strum
itertools
cargo-metadata
</details>

Also

<details>
<summary> quite a few duplicates </summary>

ahash (0.2.19, 0.3.8)
ansi_term (0.11.0, 0.12.1)
arrayvec (0.4.12, 0.5.1)
autocfg (0.1.7, 1.0.0)
base64 (0.12.3, 0.13.0, 0.11.0)
block-buffer (0.7.3, 0.9.0)
block-cipher (0.7.1, 0.8.0)
block-padding (0.1.5, 0.2.1)
bs58 (0.3.1, 0.4.0)
bytes (0.5.6, 0.4.12)
cfg-if (1.0.0, 0.1.10)
cloudabi (0.1.0, 0.0.3)
crypto-mac (0.8.0, 0.7.0)
curve25519-dalek (2.1.0, 3.0.0)
digest (0.9.0, 0.8.1)
futures (0.1.29, 0.3.5)
futures-timer (2.0.2, 3.0.2)
generic-array (0.12.3, 0.14.3)
getrandom (0.2.0, 0.1.14)
gimli (0.22.0, 0.21.0)
h2 (0.2.6, 0.1.26)
hashbrown (0.1.8, 0.8.1, 0.6.3)
hmac (0.8.1, 0.7.1)
http (0.1.21, 0.2.1)
http-body (0.3.1, 0.1.0)
hyper (0.13.7, 0.12.35)
idna (0.2.0, 0.1.5)
itertools (0.9.0, 0.8.2)
lock_api (0.1.5, 0.3.4, 0.4.1)
lru (0.5.3, 0.4.3, 0.6.0)
miow (0.2.1, 0.3.5)
object (0.19.0, 0.20.0)
once_cell (0.1.8, 1.4.1)
opaque-debug (0.2.3, 0.3.0)
parity-wasm (0.32.0, 0.41.0)
parking_lot (0.9.0, 0.7.1, 0.10.2, 0.11.0)
parking_lot_core (0.7.2, 0.4.0, 0.8.0, 0.6.2)
paste (1.0.0, 0.1.18)
pbkdf2 (0.4.0, 0.3.0)
percent-encoding (1.0.1, 2.1.0)
pin-project (1.0.1, 0.4.22)
pin-project-internal (1.0.1, 0.4.22)
pwasm-utils (0.16.0, 0.14.0)
quick-error (1.2.3, 2.0.0)
rand (0.5.6, 0.3.23, 0.6.5, 0.7.3, 0.4.6)
rand_chacha (0.2.2, 0.1.1)
rand_core (0.4.2, 0.3.1, 0.5.1)
rand_hc (0.1.0, 0.2.0)
rand_pcg (0.1.2, 0.2.1)
scopeguard (0.3.3, 1.1.0)
semver (0.6.0, 0.10.0, 0.9.0)
send_wrapper (0.4.0, 0.2.0, 0.3.0)
sha-1 (0.9.2, 0.8.2)
sha2 (0.9.1, 0.8.2)
sha3 (0.9.1, 0.8.2)
smallvec (1.4.1, 0.6.13)
stream-cipher (0.7.1, 0.4.1)
subtle (1.0.0, 2.2.3)
tokio (0.2.22, 0.1.22)
unsigned-varint (0.5.1, 0.4.0)
url (2.1.1, 1.7.2)
want (0.2.0, 0.3.0)
wasmparser (0.59.0, 0.57.0)
winapi (0.2.8, 0.3.9)
x25519-dalek (0.6.0, 1.1.0)
</details>

I'm not saying it all should be fixed here, but there are some low-hanging fruits that would reduce the amount of dependabot PRs.

**Comment by mxinden** [comment]
In regards to the list of upgrades: I would prefer we look at each of these individually. As you mentioned they potentially have breaking changes of which likely not all can be detected at compile time.

In regards to the list of duplicates: Please correct me if I am wrong, but I don't think this can be solved with a single pull request to the Substrate repository. These duplicates emerge due to dependencies and transitive dependencies having themselves outdated dependencies.

**Review by ordian — APPROVED** [review]
The crates upgrades can be done in a separate PRs if you prefer. 
I don't think we'd have to deal with runtime breaking changes though.

**Comment by ordian** [comment]
> Please correct me if I am wrong, but I don't think this can be solved with a single pull request to the Substrate repository

Sure, but upgrading dependencies can substantially reduce duplicates, e.g. https://github.com/openethereum/parity-ethereum/pull/11467

File: Cargo.lock [hunk 1]
```diff
@@ -687,9 +674,9 @@ dependencies = [
 
 [[package]]
 name = "cc"
-version = "1.0.58"
+version = "1.0.63"
```
Comment (ordian) [inline Cargo.lock:677]: it seems that upgrade of `cc` caused some issues https://github.com/rust-rocksdb/rust-rocksdb/issues/474#issuecomment-730377875

Comment (mxinden) [reply Cargo.lock]: Thanks for the pointer @ordian!

Downgrading to `v1.0.62` seems to compile fine on the build machine. Let's see if the same is the case for our pipeline steps or whether I need to do a fresh recompile to trigger the failure.

**Comment by tomaka** [comment]
You might be able to fix the wasm-bindgen problem by putting `0.2.68` [here](https://github.com/paritytech/substrate/blob/master/bin/node/browser-testing/Cargo.toml#L15).

**Review by wheresaddie — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

File: Cargo.lock (hunk 1 above)
Comment (clearloop) [reply Cargo.lock]: https://github.com/paritytech/substrate/issues/7522

**Pull request merged**